packages/api/lambdas/sf-event-sqs-to-db-records/write-execution.js

```diff
@@ -1,69 +1,155 @@
-const log = require('@cumulus/common/log');
+const isNil = require('lodash/isNil');
+
 const {
   tableNames,
 } = require('@cumulus/db');
 const {
   getMessageExecutionArn,
+  getExecutionUrlFromArn,
   getMessageCumulusVersion,
+  getMessageWorkflowTasks,
+  getMessageWorkflowName,
+  getMessageWorkflowStartTime,
+  getMessageWorkflowStopTime,
+  getMessageExecutionOriginalPayload,
+  getMessageExecutionFinalPayload,
 } = require('@cumulus/message/Executions');
 const {
   getMetaStatus,
 } = require('@cumulus/message/workflows');
 
+const { parseException } = require('../../lib/utils');
 const Execution = require('../../models/executions');
 
 const {
   isPostRDSDeploymentExecution,
-  hasNoAsyncOpOrExists,
-  hasNoParentExecutionOrExists,
 } = require('./utils');
 
-const shouldWriteExecutionToRDS = async (
+const shouldWriteExecutionToRDS = ({
   cumulusMessage,
   collectionCumulusId,
-  knex
-) => {
+  asyncOperationCumulusId,
+  parentExecutionCumulusId,
+}) => {
   const isExecutionPostDeployment = isPostRDSDeploymentExecution(cumulusMessage);
   if (!isExecutionPostDeployment) return false;
 
-  try {
-    if (!collectionCumulusId) return false;
+  return !isNil(collectionCumulusId)
```

Comment:
A bonus of the refactor is that this method is a lot simpler now

Reply:
Probably dumb question - are collections currently required for Executions?    Didn't we make that optional so non-data oriented executions could occur?    The collection doesn't appear to be required in the API schema, and the FK isn't set to be non-nullable on the Postgres migration. 

  In this code if the call further up the stack in the lambda index.js getMessageCollectionCumulusId(cumulusMessage, knex) can return `undefined` and isNil(undefined) resolves to true. 

Same for async. 

The original call to hasNoAsyncOpOrExists would return true if there was no async op ID defined in the message *or* if there was one in the message and it was in the DB. 

That doesn't appear to be true anymore: 


```
const getMessageAsyncOperationCumulusId = async (cumulusMessage, knex) => {
  try {
    const asyncOperationId = getMessageAsyncOperationId(cumulusMessage);
    if (!asyncOperationId) {
      throw new Error('Could not find async operation in message');
    }
    const asyncOperation = await knex(tableNames.asyncOperations).where({
      id: asyncOperationId,
    }).first();
    if (!isRecordDefined(asyncOperation)) {
      throw new Error(`Could not find async operation with id ${asyncOperationId}`);
    }
    return asyncOperation.cumulus_id;
  } catch (error) {
    log.error(error);
    return undefined;
  }
};

```

I'm sure I'm missing/not thinking about something right here.... 🤔

Reply:
Your concerns were very valid. The logic has been refactored to account for the fact that we need to know both whether there is a reference to an object in the message **and** whether that object exists in the Postgres database.